### PR TITLE
doc overhaul step 1: simplify index and group deprecated files

### PR DESCRIPTION
**Pull request by alvyjudy**
This is the first step on #2093 . Quite a few links might be broken for the time being but I would like to make sure I'm on the right track before moving forward

## Summary of changes
1) simplified top index into three entires: user guide, development guide, and deprecated (depth is now 1, before I can clean up the entries for sub index)
2) renamed doc for python eggs (formats.txt --> python_egg.txt)
3) move python_eggs, easy_intall, python2to3 docs into dedicated folder (deprecated)
4) create index inside "deprecated" folder and reference it in the top index

### Pull Request Checklist
- [ ] Changes have tests
- [x] News fragment added in changelog.d.

### CODE REVIEW

**Comment by alvyjudy** [comment]
If possible, I would like to have this PR merged into a separate branch rather than master because it is an ongoing process and might (and already) break some links until it's completed:)

File: docs/index.txt [hunk 1]
```diff
@@ -14,12 +14,8 @@ designed to facilitate packaging Python projects, where packaging includes:
...
-   :maxdepth: 2
+   :maxdepth: 1
 
-   setuptools
-   pkg_resources
-   python3
-   development
-   roadmap
-   Deprecated: Easy Install <easy_install>
-   history
```
Comment (webknjaz) [inline docs/index.txt:25]: Was the removal of the changelog deliberate?

Comment (jaraco) [reply docs/index.txt:25]: No. The changelog should be retained, and it perhaps should even have a forwarding link so historical links can still reach it.

Comment (webknjaz) [reply docs/index.txt:25]: FWIW I've sent a few recovery PRs like https://github.com/pypa/setuptools/pull/2425

**Pull request merged**